### PR TITLE
feat: introduce metrics

### PR DESCRIPTION
**Pull request by zsluedem**

### CODE REVIEW

File: bin/silius/src/bundler.rs [hunk 1]
```diff
@@ -492,12 +529,18 @@ where
     M: Middleware + Clone + 'static,
 {
     if let Some(chain) = chain {
-        if !CHAINS.contains(&chain) {
-            warn!("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
-            warn!("Chain {:?} is not officially supported yet! You could possibly meet a lot of problems with silius. Use at your own risk!!", chain);
-            warn!("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
+        match chain {
```
Comment (Vid201) [inline bin/silius/src/bundler.rs:532]: What's the reason behind changing this?

File: bin/silius/src/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,480 @@
+use async_trait::async_trait;
```
Comment (Vid201) [inline bin/silius/src/metrics.rs:1]: Should we move this to new crate - metrics?

**Review by Vid201 — COMMENTED** [review]
GJ!! left some comments

File: crates/primitives/src/label.rs
[the remark this answers is not in the document]
Comment (zsluedem) [reply crates/primitives/src/label.rs:1]: sry. Some leftovers.

File: bin/silius/src/bundler.rs (hunk 1 above)
Comment (zsluedem) [reply bin/silius/src/bundler.rs:532]: sry. These are some leftovers.

File: bin/silius/src/metrics.rs (hunk 2 above)
Comment (zsluedem) [reply bin/silius/src/metrics.rs:1]: Do you mean a new workspace crate?

Comment (Vid201) [reply bin/silius/src/metrics.rs:1]: Yes. What do you think, does it make sense?

Comment (zsluedem) [reply bin/silius/src/metrics.rs:1]: Ok. Make sense

File: bin/silius/src/cli/args.rs [hunk 3]
```diff
@@ -290,6 +295,25 @@ impl P2PArgs {
         }
     }
 }
+
+#[derive(Clone, Debug, Parser, PartialEq)]
+pub struct MetricArgs {
```
Comment (Vid201) [inline bin/silius/src/cli/args.rs:300]: MetricsArgs

**Review by Vid201 — APPROVED** [review]
Looking good

Left just one small nitpick

**Pull request merged**